indexing-service/src/main/java/io/druid/indexing/overlord/TaskRunner.java

```diff
@@ -39,7 +40,16 @@
    * Some task runners can restart previously-running tasks after being bounced. This method does that, and returns
    * the list of tasks (and status futures).
    */
-  public List<Pair<Task, ListenableFuture<TaskStatus>>> restore();
+  List<Pair<Task, ListenableFuture<TaskStatus>>> restore();
+
+  /**
+   * Register a listener with this task runner. On registration, the listener will get events corresponding to the
+   * current state of known tasks.
+   *
+   * @param listener the listener
+   * @param executor executor to run callbacks in
+   */
+  void registerListener(TaskRunnerListener listener, Executor executor);
```

Comment:
suggest documenting that Executor must provide its own ordering guarantees on execution.